frontend/js/redux/utils/checkStatus.js

```diff
@@ -1,13 +1,16 @@
 'use strict';
 
 export default function checkStatus(response) {
-    if ( (response.status || response.statusCode) >= 200 &&
-       (response.status || response.statusCode) < 300) {
-        return response;
-    }
-    else {
-        const error = new Error(response.statusText);
-        error.response = response;
-        throw error;
-    }
+    return new Promise((resolve, reject) => {
+        if ( (response.status || response.statusCode) >= 200 &&
+        (response.status || response.statusCode) < 300) {
```

Comment:
could be shortened to just < 300 maybe but this is cool. 1xx status codes probably ok